File: PROJETO/src/main/java/com/example/projeto/model/MotoDefeituosa.java
package com.example.projeto.model;


import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MotoDefeituosa {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Nome da marca é obrigatório")
    private String nomeMarca;

    @NotBlank(message = "Nome da moto é obrigatório")
    private String nomeMoto;

    @NotBlank(message = "Placa é obrigatório")
    private String placa;

    private String descricao;
    
    @Enumerated(EnumType.STRING)
    private TipoDefeito tipoDefeito;

    @ManyToOne
    private HistoricooOrganizacao historico;
}

File: Model/MotoDefeituosa.cs
using System.ComponentModel.DataAnnotations;
using MotoScript_Challange.Model;

namespace MotoScript_Challange.Model
{
    public class MotoDefeituosa
    {
        [Key]
        public long Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string nomeMarca { get; set; }

        [Required]
        [MaxLength(200)]
        public string nomeMoto { get; set; }

        [Required]
        [MaxLength(200)]
        public string placa { get; set; }

        [Required]
        public string descricao { get; set; }

        [Required]
        public Enums.TipoDefeito TipoDefeituoso { get; set; }  // setter público

        public MotoDefeituosa()
        {
            Id = 0;
            nomeMarca = string.Empty;
            nomeMoto = string.Empty;
            placa = string.Empty;
            descricao = string.Empty;
        }
    }
}
